modules/widgets/customizer-utils.js

```diff
@@ -0,0 +1,77 @@
+/* global gapi, FB, twttr */
+
+/**
+ * Utilities to work with widgets in Customizer.
+ */
+
+/**
+ * Checks whether this Customizer supports partial widget refresh.
+ * @returns {boolean}
+ */
+wp.customizerHasPartialWidgetRefresh = function() {
+	return 'object' === typeof wp && 'function' === typeof wp.customize && 'object' === typeof wp.customize.selectiveRefresh && 'object' === typeof wp.customize.widgetsPreview && 'function' === typeof wp.customize.widgetsPreview.WidgetPartial;
+};
+
+/**
+ * Verifies that the placed widget ID contains the widget name.
+ * @param {object} placement
+ * @param {string} widgetName
+ * @returns {*|boolean}
+ */
+wp.isJetpackWidgetPlaced = function( placement, widgetName ) {
+	var regex = new RegExp( '^' + widgetName + '-\\d+$', 'g' );
```

Comment:
I've kept the regex that Weston wrote. I think we can certainly check that only the widget name is in the string but it doesn't do any harm to check that it ends in a number as well.